core/config/config.go

```diff
@@ -0,0 +1,164 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pelletier/go-toml"
+	"github.com/tidwall/sjson"
+	"gopkg.in/yaml.v2"
+)
+
+type Handler struct {
+	cfg      interface{}
+	filePath string
+	fileExt  string
+}
+
+func New(path string, cfg interface{}) *Handler {
```

Comment:
```suggestion
func NewHandler(path string, cfg interface{}) *Handler {
```
Generally constructors in go are called NewThing(...). This is nit-picky I know.